test/jdk/javax/swing/JTable/bug4118307.java

```diff
@@ -0,0 +1,184 @@
+    static final Object[][] data = {
+            {5, 3.14},
+            {10, 2.71},
+            {70, 3.14},
+            {200, 2.71},
+    };
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(250);
+            SwingUtilities.invokeAndWait(() -> createTestUI());
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() -> {
+                tableLoc = tbl.getLocationOnScreen();
+                p = tbl.getCellRect(0, 0, true).getLocation();
+            });
+            robot.waitForIdle();
+
+            robot.mouseMove(tableLoc.x + p.x + 10, tableLoc.y + p.y + 10);
+            robot.waitForIdle();
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() ->
+                    p = tbl.getCellRect(1, 1, true).getLocation());
+            robot.waitForIdle();
+
+            robot.mouseMove(tableLoc.x + p.x + 10, tableLoc.y + p.y + 10);
+            robot.waitForIdle();
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() ->
+                    p = tbl.getCellRect(1, 0, true).getLocation());
+            robot.waitForIdle();
+
+            robot.mouseMove(tableLoc.x + p.x + 10, tableLoc.y + p.y + 10);
+            robot.waitForIdle();
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(5000);
+
+            if (isFail()) {
+                throw new RuntimeException("Test Failed.");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    public static void createTestUI() {
+        frame = new JFrame("bug4118307");
+        MyTableModel myModel = new MyTableModel();
+        tbl = new MyTable(myModel);
+        JScrollPane sp = new JScrollPane(tbl);
+        frame.add(sp, BorderLayout.CENTER);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    static synchronized void setFail() {
+        flag = true;
+    }
+
+    static synchronized boolean isFail() {
+        return flag;
+    }
```

Comment:
These can be removed once flag is made volatile and flag can directly be used in the test